### PR TITLE
Add new Component Date-Range to the statistics consumption 

### PR DESCRIPTION
**Pull request by mahdibenromdhane**
If it's validate technically/functionally, we can integrate it, in the rest of the project.

### CODE REVIEW

File: src/app/pages/statistics/consumption/statistics-consumption.component.ts [hunk 1]
```diff
@@ -57,10 +59,8 @@ export class StatisticsConsumptionComponent implements OnInit {
   public ngOnInit(): void {
     let filterDef: TableFilterDef;
 
-    filterDef = new StartDateFilter().getFilterDef();
-    this.allFiltersDef.push(filterDef);
-
-    filterDef = new EndDateFilter().getFilterDef();
+    filterDef = new DateRangeFilter().getFilterDef();
+    filterDef.timePicker24Hour = this.language !== 'en';
```
Comment (LucasBrazi06) [inline src/app/pages/statistics/consumption/statistics-consumption.component.ts:63]: Kind of a hack, locale should be passed to the date/time picker and it should handle the way the hours are formatted or find another way the set the 24 hours (check moment API ?)

File: src/app/pages/statistics/consumption/statistics-consumption.component.ts [hunk 2]
```diff
@@ -90,6 +90,10 @@ export class StatisticsConsumptionComponent implements OnInit {
     this.selectedDateFrom = date;
```
Comment (LucasBrazi06) [inline src/app/pages/statistics/consumption/statistics-consumption.component.ts:90]: Is it still used?

File: src/app/pages/statistics/consumption/statistics-consumption.component.ts [hunk 3]
```diff
@@ -90,6 +90,10 @@ export class StatisticsConsumptionComponent implements OnInit {
...
   }
 
+  public dateRangeChange(date: any) {
+    this.selectedDateRange = date;
+  }
+
   public dateToChange(date: Date) {
```
Comment (LucasBrazi06) [inline src/app/pages/statistics/consumption/statistics-consumption.component.ts:97]: Is it still used?

File: src/app/pages/statistics/shared/statistics-filters.component.ts [hunk 4]
```diff
@@ -372,6 +412,34 @@ export class StatisticsFiltersComponent implements OnInit {
     this.export.emit();
   }
 
+  // set Date Filter to corresponding year
+  private setDateRangeFilterYear(init = false): void {
+    this.statFiltersDef.forEach((filterDef: StatisticsFilterDef) => {
+      if (filterDef.type === FilterType.DATE_RANGE) {
+        if (init) {
+          this.initDateRange = true;
+          filterDef.timePicker24Hour = !(this.language === 'en');
```
Comment (LucasBrazi06) [inline src/app/pages/statistics/shared/statistics-filters.component.ts:421]: Same remark as above

File: src/app/shared/table/filters/date-range-filter.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  constructor() {
+    super();
+    // Define filter
+    const filterDef: TableFilterDef = {
+      id: 'dateRange',
+      httpId: 'DateRange',
+      type: FilterType.DATE_RANGE,
+      name: 'general.search_date',
+      class: 'col-sm-6 col-md-4 col-lg-3 col-md-3',
+      timePicker24Hour: true,
```
Comment (LucasBrazi06) [inline src/app/shared/table/filters/date-range-filter.ts:16]: Should be resolved with user's locale

File: src/app/shared/table/filters/date-range-filter.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { FilterType, TableFilterDef } from 'app/types/Table';
```
Comment (LucasBrazi06) [inline src/app/shared/table/filters/date-range-filter.ts:1]: Rename date-range-filter.ts -> date-range-table-filter.ts
Follow what is already in place.
All filter class names should end with *table-filter.ts

File: src/app/types/Table.ts [hunk 7]
```diff
@@ -40,8 +41,22 @@ export interface TableFilterDef {
   multiple?: boolean;
   exhaustive?: boolean;
   cleared?: boolean;
+  singleDatePicker?: boolean;
```
Comment (LucasBrazi06) [inline src/app/types/Table.ts:44]: This is specific date/time properties, they should be in this type I think but in a sub-type like DateRangeTableFilterDef.
What is weird is that we didn't need it for the Date filter why now?

File: src/assets/scss/views/statistics/_statistics.scss [hunk 8]
```diff
@@ -0,0 +1,31 @@
+.md-drppicker {
```
Comment (LucasBrazi06) [inline src/assets/scss/views/statistics/_statistics.scss:1]: This should be put in a dedicated files for the date/time picker and imported globally

File: src/app/types/Table.ts (hunk 7 above)
Comment (mahdibenromdhane) [reply src/app/types/Table.ts:48]: the date range provide more properties to control the component that we can use to customize. ( it's not mandatory)

**Comment by mahdibenromdhane** [comment]
> Here is my feedbacks:
> 
> 1\ Input field is not visible when date selector is shown
> image
> 
> 2\ Icon is missing
> 
> image
> 
> 3\ Only when icon is clicked the selector should be displayed
> 
> 4\ If I click in the input field, I should be able to change the date/time manually

Done, but the popup will still open while we type

File: src/app/pages/statistics/shared/statistics-filters.component.ts [hunk 9]
```diff
@@ -124,11 +143,50 @@ export class StatisticsFiltersComponent implements OnInit {
...
     this.filterParams = this.buildFilterValues();
     this.filters.emit(this.filterParams);
     this.update.emit(true);
   }
 
+  public dateRangeChange(filterDef: StatisticsFilterDef, event): void {
+    if (filterDef.type === 'date-range' && event.hasOwnProperty('startDate') && event.hasOwnProperty('endDate')) {
+      filterDef.currentValue = event ? event : null;
+    } else {
+      const dateRange = event.currentTarget.value;
```
Comment (LucasBrazi06) [inline src/app/pages/statistics/shared/statistics-filters.component.ts:156]: Why the else here?
Manual update by user?

File: src/assets/scss/views/shared/_date-picker.scss [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    .md-drppicker td.available:hover,
+    .md-drppicker th.available:hover {
+      background-color: rgba(0, 0, 0, 0.04);
+      color: rgba(0, 0, 0, 0.87); }
+    .md-drppicker td.off, .md-drppicker td.off.in-range, .md-drppicker td.off.start-date, .md-drppicker td.off.end-date {
+      background-color: white;
+      border-color: white;
+      color: rgba(0, 0, 0, 0.38); }
+    .md-drppicker td.in-range {
+      background-color: #eeeeee;
```
Comment (LucasBrazi06) [inline src/assets/scss/views/shared/_date-picker.scss:17]: Do not use hardcoded colors, there is a theme.
Use SCSS variable like this:
```
    .charger-connector-charging {
      background-color: mat-color($mat-blue, 600);
      border: 2px solid darken(mat-color($mat-blue, 600), 15%);
    }

```

Comment (LucasBrazi06) [reply src/assets/scss/views/shared/_date-picker.scss:17]: Update the whole _date-picker.scss file

File: src/app/pages/statistics/shared/statistics-filters.component.ts (hunk 9 above)
Comment (mahdibenromdhane) [reply src/app/pages/statistics/shared/statistics-filters.component.ts:156]: Yes for the manual update

**Pull request merged**